### PR TITLE
[Feature] support vectorized compaction

### LINKED ISSUE #8445: [Feature] [vectorized]support vectorized compaction
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description
currently, compaction is not vectorized, to improve the compaction performance, supporting vectorized compaction is needed

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #8445 
Btw, this PR is a follow-up for PR [8438](https://github.com/apache/incubator-doris/pull/8438)

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
I will attach a performance report later.

**Comment by Gabriel39** [comment]
I finish the performance test and add the progress and performance report in my own repo. Pls refer to https://github.com/Gabriel39/doris_compaction_perf/tree/main/report .

**Comment by Gabriel39** [comment]
cc @morningman @HappenLee @yiguolei

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 1]
```diff
@@ -99,6 +99,71 @@ Status BetaRowsetWriter::init(const RowsetWriterContext& rowset_writer_context)
...
+        block_row_num > segment_capacity_in_rows) {
+        size_t segment_max_row_num;
+        size_t input_row_num;
+        do {
+            assert(row_offset < block_row_num);
+            segment_max_row_num =
+                    std::max(std::min((size_t)segment_capacity_in_bytes / row_avg_size_in_bytes,
+                             (size_t)segment_capacity_in_rows), (size_t)1);
+            input_row_num = std::min(segment_max_row_num, block_row_num - row_offset);
+            assert(input_row_num > 0);
```
Comment (morningman) [inline be/src/olap/rowset/beta_rowset_writer.cpp:140]:
```suggestion
            CHECK(input_row_num > 0);
```

File: be/src/olap/rowset/beta_rowset_writer.cpp [hunk 2]
```diff
@@ -99,6 +99,71 @@ Status BetaRowsetWriter::init(const RowsetWriterContext& rowset_writer_context)
...
+        RETURN_NOT_OK(_create_segment_writer(&_segment_writer));
+        refresh_segment_capacity();
+    }
+
+    if (block_size_in_bytes > segment_capacity_in_bytes ||
+        block_row_num > segment_capacity_in_rows) {
+        size_t segment_max_row_num;
+        size_t input_row_num;
+        do {
+            assert(row_offset < block_row_num);
```
Comment (morningman) [inline be/src/olap/rowset/beta_rowset_writer.cpp:135]:
```suggestion
            CHECK(row_offset < block_row_num);
```

**Review by morningman — APPROVED** [review]
LGTM

**Pull request closed without merging**